cms-publish-rest/src/main/java/se/simonsoft/cms/publish/rest/PublishResource.java

```diff
@@ -259,10 +260,34 @@ public void write(OutputStream os) throws IOException, WebApplicationException {
 		};
 		
 		return Response.ok(stream, MediaType.APPLICATION_OCTET_STREAM)
-				.header("Content-Disposition", "attachment; filename=" + storageFactory.getNameBase(itemId, null) + ".zip")
+				.header("Content-Disposition", "attachment; filename=" + getFilename(items, publication, releaseItem) + ".zip")
 				.build();
 	}
 	
+	private String getFilename(List<CmsItem> items, String publication, CmsItem releaseItem) {
+		long rev = 0;
+		for (CmsItem item: items) {
+			long number = item.getId().getPegRev();
+			if (rev < number) {
+				rev = number;
+			}
+		}
+		
+		String releaseLabel = new CmsItemPublish(releaseItem).getReleaseLabel();
+		
+		if (releaseLabel == null) {
+			throw new IllegalStateException("Item must be a release");
```

Comment:
"The Release does not have a Release Label property."

Reply:
fixed